src/inc/til/u8u16convert.h

```diff
@@ -76,32 +96,148 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
-                            if ((*backIter & _cmpMasks.at(sequenceLen)) != _cmpOperands.at(sequenceLen))
+                            --backIter;
+                            // If Lead Byte found
+                            if ((*backIter & _Utf8BitMasks::MaskContinuationByte) > _Utf8BitMasks::IsContinuationByte)
+                            {
+                                // If the Lead Byte indicates that the last bytes in the string is a partial UTF-8 code point then cache them:
+                                //  Use the bitmask at index `sequenceLen`. Compare the result with the operand having the same index. If they
+                                //  are not equal then the sequence has to be cached because it is a partial code point. Otherwise the
+                                //  sequence is a complete UTF-8 code point and the whole string is ready for the conversion to hstring.
+                                if ((*backIter & _cmpMasks.at(sequenceLen)) != _cmpOperands.at(sequenceLen))
+                                {
+                                    std::move(backIter, _buffer.end(), _partials.begin());
+                                    remainingLength -= sequenceLen;
+                                    _partialsLen = sequenceLen;
+                                }
+
+                                break;
+                            }
+                        }
+                    }
+                }
+                else // ANSI codepages
+                {
+                    if (_cpInfo.MaxCharSize == 2u && !!_cpInfo.LeadByte[0]) // DBCS codepages
+                    {
+                        auto foundLeadByte{ false };
+
+                        // Start at the beginning of the string (or the second bvte if a lead byte has been cached in the previous call) and scan forward,
+                        // keep track when it encounters a lead byte, and treat the next byte as the trailing part of the same character.
+                        std::for_each(_buffer.cbegin(), _buffer.cend(), [&](const auto& ch) {
+                            if (foundLeadByte)
+                            {
+                                foundLeadByte = false; // because it's the trailing part
+                            }
+                            else
+                            {
+                                // We use our own algorithm because IsDBCSLeadByteEx() supports only a subset of DBCS codepages.
+                                const auto uCh{ gsl::narrow_cast<byte>(ch) };
+#pragma warning(push)
+#pragma warning(suppress : 26482) // Suppress bounds.2 check for indexing with constant expressions.
+#pragma warning(suppress : 26446) // Suppress bounds.4 check for subscript operator.
+                                for (int idx{}; _cpInfo.LeadByte[idx] != 0; idx += 2) // OK because the LeadByte array is guaranteed to end with two 0 bytes.
+#pragma warning(pop)

```

Comment:
This is what `til::at` is for. It suppresses those things for you.

Reply:
@miniksa If you scroll up you may recall that I was desparate not being able to find a way to access a C-array without suppressing warnings all the time.

About `til::at`:
> The current implementation limits it to class types that have a size() method. IMO it should work for most to all types that support the subscript operator, such like random-access iterators, C-arrays, and pointers. We should accept signed index types because iterators and pointers support negative offsets. And we should avoid to cast the passed index to `ptrdiff_t` or `size_t `because the expected type size might be smaller. E.g. `winrt::hstring::size_type` is always `uin32_t`.

I tried to update `til::at` but was fed up to fail. Do you want me to update it? I'd be happy to try again.

Reply:
Sorry, I didn't see it or forgot. If at isn't good enough, I'm happy to see it improve if that isn't too much work such that others can use it to solve the same problem going forward. There is also a gsl::at, I think, which might work with signed types already...

Reply:
Also `gsl::at` doesn't handle arrays. I'll do my best to provide a working set of overloads for `til::at` 😀

Reply:
@miniksa Do you have any hint why the static analysis still fails?
The screenshot shows that at least the IntelliSense confirms that the overload for an array will be called. However, the analysis here still complains about array to pointer decay while I don't get any warning in my local tests.
![image](https://user-images.githubusercontent.com/46659645/76127735-ba4c8700-6002-11ea-8bcb-63a3e641ac55.png)

Reply:
Eventually I removed the overloads for arrays since the static analysis doesn't seem to understand that an array decay is not necessary if it takes the array overload in consideration.
Using a combination of `*` and `&` I'm able to get the pointer to the first element in the array without triggering a warning. The updated `til::at` function is able to handle pointers properly.
That's how I call it
`til::at(&(*_cpInfo.LeadByte), idx)`
Doesn't look nice, I know.